nextcord/automod.py

```diff
@@ -0,0 +1,366 @@
+        self.guild_id: int = int(data["guild_id"])
+        self.name: str = data["name"]
+        self.creator_id: int = int(data["creator_id"])
+        self.event_type: EventType = try_enum(EventType, data["event_type"])
+        self.trigger_type: TriggerType = try_enum(TriggerType, data["trigger_type"])  # same
+        self.enabled: bool = data["enabled"]
+        self.exempt_role_ids: List[int] = [int(exempt_role) for exempt_role in data["exempt_roles"]]
+        self.exempt_channel_ids: List[int] = [
+            int(exempt_channel) for exempt_channel in data["exempt_channels"]
+        ]
+        self.keyword_filters: Optional[List[str]] = None
+        self.presets: Optional[List[KeywordPresetType]] = None
+
+        trigger_metadata: TriggerMetadataPayload = data["trigger_metadata"]
+
+        self.keyword_filters = trigger_metadata.get("keyword_filter")
+
+        self.actions: List[AutoModerationAction] = []
+        self.presets = [try_enum(KeywordPresetType, preset) for preset in trigger_metadata["presets"]] if trigger_metadata.get("presets") is not None else None  # type: ignore
+
+        for action in data["actions"]:
+            self.actions.append(AutoModerationAction(data=action))
+
+    def __repr__(self):
+        attrs = (
+            ("id", self.id),
+            ("guild", self.guild),
+            ("guild_id", self.guild_id),
+            ("name", self.name),
+            ("creator_id", self.creator_id),
+            ("event_type", self.event_type),
+            ("trigger_type", self.trigger_type),
+            ("enabled", self.enabled),
+            ("exempt_role_ids", self.exempt_role_ids),
+            ("exempt_channel_ids", self.exempt_channel_ids),
+            ("keyword_filters", self.keyword_filters),
+            ("presets", self.presets),
+            ("actions", self.actions),
+        )
+        inner = " ".join("%s=%r" % t for t in attrs)
+        return f"<AutoModerationRule {inner}>"
+
+    async def delete(self):
+        """|coro|
+
+        Delete this auto moderation rule.
+        """
+        await self._state.http.delete_automod_rule(guild_id=self.guild.id, rule_id=self.id)
+
+    @property
+    def creator(self) -> Optional[Member]:
+        """Optional[:class:`Member`]: The member that created this rule."""
+        return self.guild.get_member(self.creator_id)
+
+    @property
+    def exempt_roles(self) -> List[Role]:
+        """List[:class:`Role`]: A list of role that will not be affected by this rule. `[]` if not set."""
+        return [self.guild.get_role(exempt_role_id) for exempt_role_id in self.exempt_role_ids]  # type: ignore -- they can't be None.
+
+    @property
+    def exempt_channels(self) -> List[GuildChannel]:
+        """List[:class:`GuildChannel`]: A list of channels that will not be affected by this rule. `[]` if not set."""
+        return [self.guild.get_channel(exempt_channel_id) for exempt_channel_id in self.exempt_channel_ids]  # type: ignore -- same
```

Comment:
Its possible `--` does the same, but for some type checkers you need the extra comment for anything extra

```
# type: ignore # same
```